source/config.mts

```diff
@@ -0,0 +1,72 @@
+import path from "path";
+import fs from "fs/promises";
+import { compile } from "./main.coffee";
+
+const findInDir = async function (dirPath: string): Promise<string | undefined> {
+  const dir = await fs.opendir(dirPath);
+  for await (const entry of dir) {
+    if (
+      entry.isDirectory() &&
+      entry.name === ".config"
+    ) {
+      // scan for ./civet.json as well as ./.config/civet.json
+      const found = await findInDir(
+        path.join(dirPath, entry.name)
+      );
+      if (found) {
+        return found;
+      }
+    }
+    if (entry.isFile()) {
+      const name = entry.name.replace(/^\./, ""); // allow both .civetconfig.civet and civetconfig.civet
+      if ([
+        "🐈.json",
+        "🐈.civet",
+        "civetconfig.json",
+        "civetconfig.civet",
+      ].includes(name)) {
```

Comment:
`name in ['config.civet', 'civet.json', 'civetconfig.civet', 'civetconfig.json']` has a more efficient compilation in CoffeeScript than this. But I guess we could do even better with a `Set` (hoisted to global scope).